File: week2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    float l = (letters / (float) words) * 100;
    float s = (sentences / (float) words) * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i])) // counts all alphanumerical characters
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int word = 1;

    for (int w = 0; w < strlen(text); w++)
    {
        if (isspace(text[w])) // If there is a space, count +1 word
        {
            word++;
        }
    }

    return word;
}

int count_sentences(string text)
{
    int sentence = 0;
    for (int s = 0; s < strlen(text); s++)
    {
        if (text[s] == '.' || text[s] == '!' ||
            text[s] == '?') // If there is a punctuation, count +1 sentence
        {
            sentence++;
        }
    }

    return sentence;
}

File: week6/readability.py
from cs50 import get_string

# Prompt user for text
text = get_string("Text: ")

# Count number of letters, words and sentences

letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if text[i].isalpha():
        letters += 1
    elif text[i].isspace():
        words += 1
    elif text[i] in [".", "!", "?"]:
        sentences += 1

# Calculate Coleman-Liau index

l = letters / words * 100
s = sentences / words * 100

index = round(0.0588 * l - 0.296 * s - 15.8)

# Print Grade

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")
